troposphere/__init__.py

```diff
@@ -84,12 +131,35 @@ def depends_on_helper(obj):
     if isinstance(obj, AWSObject):
         return obj.title
     elif isinstance(obj, list):
-        return list(map(depends_on_helper, obj))
+        return list(map(depends_on_helper, cast(List[object], obj)))
     return obj
 
 
+__BaseAWSObjectTypeVar = TypeVar("__BaseAWSObjectTypeVar", bound="BaseAWSObject")
+
+
 class BaseAWSObject:
-    def __init__(self, title, template=None, validation=True, **kwargs):
+
+    attributes: List[str]
+    dictname: Optional[str]
+    do_validation: bool
+    properties: Dict[str, Any]
+    propnames: Set[str]
+    props: ClassVar[
+        Dict[str, Tuple[Union[Tuple[type, ...], type, Callable[[Any], Any]], bool]]
+    ] = {}
+    resource: Dict[str, Any]
+    resource_type: Optional[str]
+    template: Optional[Template]
+    title: Optional[str]
+
+    def __init__(
+        self,
+        title: Optional[str],
+        template: Optional[Template] = None,
```

Comment:
@michael-k _technically_ yes according to PEP 484. However, for strict type checking (pyright specifically) it expects it to be explicitly stated as `Optional`. I'm not sure how mypy handles it - I stopped using it some time ago due to it having too many issues (that took too long to be fixed) and being incredibly slow.

From [pyright's docs](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#type-check-diagnostics-settings):
> **strictParameterNoneValue** [boolean]: PEP 484 indicates that when a function parameter is assigned a default value of None, its type should implicitly be Optional even if the explicit type is not. When enabled, this rule requires that parameter type annotations use Optional explicitly in this case. The default value for this setting is 'true'.

Not having the explicit `Optional` in a library like this can cause issues for those that want to use strict type checking forcing them to allow implicit `Optional`. Troposphere is one of the libraries that forced me to disable **strictParameterNoneValue** in some of my projects and is why it's disabled in the configuration I copy/pasted in this PR - I forgot to remove it. I opted to be explicit in adding my annotations to avoid the potential of causing issues for others.